src/foam/core/FObject.java

```diff
@@ -302,7 +302,62 @@ default FObject copyFrom(FObject obj) {
           if ( p2 != null ) {
             if ( p2.isSet(obj) ) p.set(this, p2.get(obj));
           }
-        } catch (ClassCastException ignore) {}
+        } catch (ClassCastException ignore) {
+          System.err.println("FObject.copyFrom "+p.getName()+" "+ignore.getMessage());
+        }
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Similar to copyFrom, with recursion on nested FObjects, only setting
+   * isSet=true properties.
+   */
+  default FObject overlay(FObject obj) {
+    return overlay_(obj, new java.util.HashSet());
+  }
+
+  default FObject overlay_(FObject obj, java.util.Set visited) {
+    int code = obj.hashCode();
+    if ( visited.contains(code) ) {
+      return this;
+    } else {
```

Comment:
No need for 'else' if you've already returned in the 'if'.

Reply:
Left over from debugging, simplified.